### DISCUSSION TITLE
global rejectOnNotFound with PrismaClient query throw type null..is there any solution to do not throw type null

### DISCUSSION THREAD

**Discussion by itezjaehyeon** [prisma/prisma#12127] [Q&A]
### version : 2.23.0

```ts
export const prisma = new PrismaClient({
    log: [
        {
            emit: "event",
            level: "query",
        },
    ],
    rejectOnNotFound: {
        findUnique: ()=>errors.etc("not found..."),
    },
});
```
`const result = await ctx.prisma.post.findUnique({ where: { id: 1 } });`
result type is `post | null` 
i want to make it `post` is there any solution for this problem??

**Comment by austincrim** [comment]
Hey @itezjaehyeon 👋 ,

Good timing, we just merged a [PR](https://github.com/prisma/prisma/pull/11501) that will fix this issue in an upcoming release of the Prisma Client. Keep your eye out for the new version and update your `@prisma/client` dependency to get the fix.

Also, I noticed you are currently on a Prisma 2 version, here's an [upgrade guide](https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-3) for moving from Prisma 2 -> Prisma 3.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
